test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -1431,18 +1436,15 @@ public void help() throws Exception {
     }
   }
 
-  // TODO - evaluate this test is testing what is expected or history is working.
   @Test
   public void history() throws Exception {
-    final String table = name.getMethodName();
+    final String table = getUniqueNames(1)[0];
 
     ts.exec("history -c", true);
-    ts.exec("createtable " + table);
-    ts.exec("deletetable -f " + table);
-    // TODO - this may be passing only because method name is history.
+    ts.writeToHistory("createtable " + table);
+    ts.writeToHistory("deletetable -f " + table);
```

Comment:
Does the write to history perform an exec() - previously the test was executing a command, testing the command was successful (via the exec status) and then that appeared in the history - from a quick look I don't see where this performs the exec.  Not that these commands need tested, but is it a valid test of shell history?

Reply:
These aren't executed. I felt it was safe to leave it out from an exec call since they aren't part of this test specifically. The exec command fails to write to the history file due how things are set up with the shell. The history file is part of the reader, but the reader isn't involved with the executions since I assume it doesn't interpret it as an input command. This is why only `ts.input` would show up in the history file since it was being read. 

From a validity standpoint, the exec of the history command does work properly and the contents of the reader's history file gets displayed.